code/modules/mob/living/simple_animal/hostile/space_dragon.dm

```diff
@@ -0,0 +1,727 @@
+	var/tiredness = 0
+	/// A multiplier to how much each use of wing gust should add to the tiredness variable.  Set to 5 if the current rift is destroyed.
+	var/tiredness_mult = 1
+	/// The distance Space Dragon's gust reaches
+	var/gust_distance = 4
+	/// The amount of tiredness to add to Space Dragon per use of gust
+	var/gust_tiredness = 30
+	/// Determines whether or not Space Dragon is in the middle of using wing gust.  If set to true, prevents him from moving and doing certain actions.
+	var/using_special = FALSE
+	/// Determines whether or not Space Dragon is currently tearing through a wall.
+	var/tearing_wall = FALSE
+	/// Whether space dragon is swallowing a body currently
+	var/is_swallowing = FALSE
+	/// A list of all of the rifts created by Space Dragon.  Used for setting them all to infinite carp spawn when Space Dragon wins, and removing them when Space Dragon dies.
+	var/list/obj/structure/carp_rift/rift_list = list()
+	/// How many rifts have been successfully charged
+	var/rifts_charged = 0
+	/// Whether or not Space Dragon has completed their objective, and thus triggered the ending sequence.
+	var/objective_complete = FALSE
+	/// The innate ability to use wing gust
+	var/datum/action/innate/gust_attack/gust
+	/// The innate ability to summon rifts
+	var/datum/action/innate/summon_rift/rift
+	/// The ability to make your sprite smaller
+	var/datum/action/small_sprite/space_dragon/small_sprite
+	/// The color of the space dragon.
+	var/chosen_color
+
+/mob/living/simple_animal/hostile/space_dragon/Initialize(mapload)
+	. = ..()
+	gust = new
+	gust.Grant(src)
+	rift = new
+	rift.Grant(src)
+	small_sprite = new
+	small_sprite.Grant(src)
+	RegisterSignal(small_sprite, COMSIG_ACTION_TRIGGER, .proc/add_dragon_overlay)
+
+/mob/living/simple_animal/hostile/space_dragon/Login()
+	. = ..()
+	if(!chosen_color)
+		dragon_name()
+		color_selection()
+
+/mob/living/simple_animal/hostile/space_dragon/Life(delta_time = SSMOBS_DT, times_fired)
+	. = ..()
+	tiredness = max(tiredness - 1, 0)
+	for(var/mob/living/consumed_mob in src)
+		if(consumed_mob.stat == DEAD)
+			continue
+		playsound(src, 'sound/effects/splat.ogg', 50, TRUE)
+		visible_message("<span class='danger'>[src] vomits up [consumed_mob]!</span>")
+		consumed_mob.forceMove(loc)
+		consumed_mob.Paralyze(50)
+	if((rifts_charged == 3 || (SSshuttle.emergency.mode == SHUTTLE_DOCKED && rifts_charged > 0)) && !objective_complete)
+		victory()
+	if(riftTimer == -1)
+		return
+	riftTimer = min(riftTimer + 1, maxRiftTimer + 1)
+	if(riftTimer == (maxRiftTimer - 60))
+		to_chat(src, "<span class='boldwarning'>You have a minute left to summon the rift!  Get to it!</span>")
+		return
+	if(riftTimer >= maxRiftTimer)
+		to_chat(src, "<span class='boldwarning'>You've failed to summon the rift in a timely manner!  You're being pulled back from whence you came!</span>")
+		destroy_rifts()
+		empty_contents()
+		playsound(src, 'sound/magic/demon_dies.ogg', 100, TRUE)
+		QDEL_NULL(src)
```

Comment:
Might be a better idea to limit their abilities until they place the rift rather than just delete them. Prevents just going in for murderbone while not outright deleting slow dragons

Reply:
Their abilities are already pretty minimal, I could remove fire breath but honestly you get 6 minutes to go onto station and make a rift. I could add the depression instead of deleting since that happens when you fail regularly.

Reply:
Removing the qdel and empty_contents. Dragon will basically get nerfed hard (destroy_rifts adds the depression and prevents future rifht placing) if they don't place rifts in time instead of just straight up deleted.